### PR TITLE
Implement interp

### PR DESCRIPTION
**Pull request by jeanlaroche**
Re-opening this after cleanup of merge and so on.

### CODE REVIEW

File: pythran/tables.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 from pythran import interval
 from functools import reduce
 
-if sys.version_info.major == 3:
+if sys.version_info.major == 3:
```
Comment (serge-sans-paille) [inline pythran/tables.py]: useless diff

File: pythran/tests/test_numpy_func3.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 
 @TestEnv.module
-class TestNumpyFunc3(TestEnv):
+class TestNumpyFunc3(TestEnv):
```
Comment (serge-sans-paille) [inline pythran/tests/test_numpy_func3.py]: same here

File: pythran/pythonic/include/numpy/interp.hpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#ifndef PYTHONIC_INCLUDE_NUMPY_INTERP_HPP
```
Comment (serge-sans-paille) [inline pythran/pythonic/include/numpy/interp.hpp]: as this code is inspired from the numpy code base, you should state the original license and source used to derive this code.

File: pythran/pythonic/numpy/interp.hpp [hunk 4]
```diff
@@ -0,0 +1,126 @@
+#ifndef PYTHONIC_NUMPY_INTERP_HPP
+#define PYTHONIC_NUMPY_INTERP_HPP
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/interp.hpp]: The header should actually be here.

File: pythran/pythonic/numpy/interp_core.cpp [hunk 5]
```diff
@@ -0,0 +1,256 @@
+//
+//  interp_core.cpp
+//  From /Users/jlaroche/temp/NumpySrc/numpy/core/src/multiarray/compiled_base.c
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/interp_core.cpp]: Well, the licencing should be reproduced here.

File: pythran/pythonic/numpy/interp_core.cpp [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+//    else {
+//        rval = PyFloat_AsDouble(right);
+//        if (error_converting(rval)) {
+//            goto fail;
+//        }
+//    }
+
+// xp->dx   fp->dy  x -> dz
+
+#define PyArray_malloc malloc
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/interp_core.cpp]: avoid that and use a std::vector that you resize appropriately, that makes memory management safer

File: pythran/pythonic/numpy/interp_core.cpp [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+    }
+    //        NPY_END_THREADS;
+  } else {
+    npy_intp j = 0;
+
+    /* only pre-calculate slopes if there are relatively few of them. */
+    if (lenxp <= lenx) {
+      slopes = (npy_double *)PyArray_malloc((lenxp - 1) * sizeof(npy_double));
+      if (slopes == NULL) {
+        // goto fail;
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/interp_core.cpp]: just remove the test, the resie will throw an exception if it fails.

File: pythran/pythonic/numpy/interp_core.cpp (hunk 5 above)
Comment (jeanlaroche) [reply pythran/pythonic/numpy/interp_core.cpp]: There isn't any licensing in the original file.

**Comment by serge-sans-paille** [comment]
> There isn't any licensing in the original file.

Then just copy the header of the numpy license, the one at the top of the project?

**Comment by jeanlaroche** [comment]
Performance does not seem to be very good on this for a reason that eludes me.
```
import numpy as np
np.random.seed(0)
xi=np.random.randn(10000)
xi=np.sort(xi)
yi=np.random.randn(10000)
x=np.random.randn(100000)

%timeit np.interp(x,xi,yi)
import jean as tt
%timeit tt.test(x,xi,yi)
```
```
import numpy as np

#pythran export test(float[],float[],float[])
def test(x,xi,yi):
    return np.interp(x,xi,yi)
```

Results in linux:
```
100 loops, best of 3: 9.87 ms per loop <- numpy
100 loops, best of 3: 9.92 ms per loop <- pythran
```

Results on macos:
```
8.76 ms ± 122 µs per loop (mean ± std. dev. of 7 runs, 100 loops each) <- numpy
9.83 ms ± 126 µs per loop (mean ± std. dev. of 7 runs, 100 loops each) <- pythran
```
Adding -march=native or -fopenmp does not change anything.

**Pull request merged**